File: C#/Queue/649. Dota2 Senate.cs
/* 649. Dota2 Senate 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/dota2-senate/submissions/1387465228/ */

public class Solution {
    List<char> list = new List<char>();
    
    public string PredictPartyVictory(string senate) {

        foreach (var item in senate)
            list.Add(item);

        while (list.Count > 1)
        {
            if (list[0] == 'R')
            {
                if(list.Contains('D'))
                    removeChar('D');
                else
                    return "Radiant";
            }
            else
            {
                if (list.Contains('R'))
                    removeChar('R');
                else
                    return "Dire";
            }
        }
        
        if (list[0] == 'D')
            return "Dire";
        return "Radiant";
    }

    public void removeChar(char ch) {
        list.Remove(ch);
        list.Add(list[0]);
        list.RemoveAt(0);
    }

}

File: JavaScript/Queue/649. Dota2 Senate.js
X/* 649. Dota2 Senate 
Difficulty: Medium
Submission Link: https://leetcode.com/problems/dota2-senate/submissions/1387464690/ */

var predictPartyVictory = function(senate) {
    senate = senate.split('');
    while(senate.length) {
        var first = senate.shift();
        var len = senate.length;

        for(var i=0; i<len; i++) {
            if(first != senate[i]) {
                senate.splice(i, 1);
                senate.push(first);
                break;
            }
        }
        if(i == len) {
            return first == 'D' ? 'Dire' : 'Radiant';
        }
    }
};